File: ddss/dump.ts
import type { Sequelize } from "sequelize";
import { unparse } from "atsds-bnf";
import { Fact, Idea } from "./orm.ts";

export async function main(sequelize: Sequelize): Promise<void> {
    const ideas = await Idea.findAll();
    for (const idea of ideas) {
        console.log("idea:", unparse(idea.data));
    }

    const facts = await Fact.findAll();
    for (const fact of facts) {
        console.log("fact:", unparse(fact.data));
    }
}

File: ddss/dump.py
import asyncio
from sqlalchemy import select
from sqlalchemy.ext.asyncio import async_sessionmaker, AsyncSession
from apyds_bnf import unparse
from .orm import Facts, Ideas


async def main(session: async_sessionmaker[AsyncSession]) -> None:
    try:
        async with session() as sess:
            for i in await sess.scalars(select(Ideas)):
                print("idea:", unparse(i.data))
            for f in await sess.scalars(select(Facts)):
                print("fact:", unparse(f.data))
    except asyncio.CancelledError:
        pass
